### PR TITLE
Add selector function

### PR DESCRIPTION
**Pull request by jnicklas**
## Motivation

Most interactors use CSS selectors to locate the pertinent elements on the page. However, there are some interactors where it would be great if we could call a method to return the elements. An example would be a `Label` interactor, which would operate on the labels associated on a field. We would want to implement the interactor like this:

```js
const Label = HTML.extend('label').selector((e) => e.labels));
```

Another example is the keyboard interactor, which would really like to use `document.activeElement` as its selector.

Adding this option makes interactors even more powerful, since they are no longer constrained to just CSS selectors.

## Approach

Optionally, selector can take a function which gets as its argument the parent element, and must return a list of elements of the appropriate type.

### Alternate Designs

None

### Possible Drawbacks or Risks

Adds additional complexity

### CODE REVIEW

**Review by cowboyd — COMMENTED** [review]
I'm all for this. How should update the docs? File a ticket to integrate after the new guides are merged?

**Comment by jnicklas** [comment]
Let's wait for #127 to be merged, and we can extend this PR with an addition to the guides.

**Review by cowboyd — APPROVED** [review]

**Pull request merged**
